test/union_test.exs

```diff
@@ -23,11 +23,237 @@ defmodule AshGraphql.UnionTest do
       """
       |> Absinthe.run(AshGraphql.Test.Schema)
 
+    possible_type_names = Enum.map(data["__type"]["possibleTypes"], & &1["name"]) |> Enum.sort()
+
     assert [
-             %{"name" => "UniontypeMemberArrayBoolean"},
-             %{"name" => "UniontypeMemberString"},
-             %{"name" => "UniontypeMemberTypedStruct"}
-           ] =
-             data["__type"]["possibleTypes"]
+             "UniontypeMemberArrayBoolean",
+             "UniontypeMemberMap",
+             "UniontypeMemberRegularStruct",
+             "UniontypeMemberString",
+             "UniontypeMemberTypedStruct"
+           ] = possible_type_names
+  end
+
+  test "custom map type from union member is generated" do
+    # Test that PersonMapType (the custom type referenced by union member) exists
+    {:ok, %{data: data}} =
+      """
+      query {
+        __type(name: "PersonMapType") {
+          name
+          kind
+          fields {
+            name
+            type {
+              name
+              kind
+            }
+          }
+        }
+      }
+      """
+      |> Absinthe.run(AshGraphql.Test.Schema)
+
+    # Should find the PersonMapType
+    assert data["__type"]["name"] == "PersonMapType"
+    assert data["__type"]["kind"] == "OBJECT"
+
+    # Should have the expected fields from PersonMap constraints
```

Comment:
can we remove the fluffy comments 🙏

Reply:
ugh, I am so used to seeing them I don't even notice them anymore :P 
they are gone now!